app/controllers/targets_controller.rb

```diff
@@ -13,12 +15,17 @@ def create
     if @target.persisted?
       redirect_to targets_path
     else
-      flash.now[:alert] = t(:targetFail)
+      flash.now[:alert] = t(:target_fail)
       render :new
     end
   end
 
   def target_params
     params.require(:target).permit(:topic, :title, :length, :latitude, :longitude)
   end
+
+  def load_topics
+    @topics = Target::TOPICS
```

Comment:
Avoid using `before_action` or other callbacks like this to load information, this are most commonly used to run some validation or authentication. To access some information that you may need in a view, you can define a helper method like this:

````
helper_method :topics # in the top of the controller

private

def topics
 @topics ||= Target::TOPICS # use ||=, it will assign Target::TOPICS only if @topics is not already defined
end
```

Then you can access the topics by just calling topics (without @) 
https://apidock.com/rails/ActionController/Helpers/ClassMethods/helper_method